### PR TITLE
Lazymaths: add min, max and median methods

### PR DESCRIPTION
**Pull request by tschoonj**
as well as some other minor changes...

### CODE REVIEW

**Comment by tschoonj** [comment]
@PeterC-DLS Something wrong with Travis?

**Comment by jonahkichwacoders** [comment]
@tschoonj Problem is not Travis, but download.eclipse.org: https://dev.eclipse.org/mhonarc/lists/cross-project-issues-dev/msg14643.html

**Comment by IanMayo** [comment]
It looks like the downloads server is back up now.

**Comment by tschoonj** [comment]
Ok, could you restart the build please?

Thanks!

**Comment by tschoonj** [comment]
`SliceNDIterator` is now being used.

About the multiple axes: though I could easily add support for that, there is currently no counterpart for that in `Dataset`. Thoughts?

**Comment by PeterC-DLS** [comment]
You can add that to `LazyMaths` by calling max/min multiple times on (sort the axes in increasing order then iterate from the highest). For `Dataset` itself, add an issue.

**Comment by tschoonj** [comment]
@PeterC-DLS done!

File: org.eclipse.january/src/org/eclipse/january/dataset/LazyMaths.java [hunk 1]
```diff
@@ -70,34 +72,53 @@ public double perform(IDataset oneDSlice) {
...
+				axes[axisIndex] += rank;
+			if (axes[axisIndex] < 0 || axes[axisIndex] >= rank) {
+				logger.error(INVALID_AXIS_ERROR);
+				throw new IllegalArgumentException(INVALID_AXIS_ERROR);
+			}
 		}
-
 		final int[] oldShape = data.getShape();
-		final int[] newShape = new int[rank-1];
+		final int[] newShape = new int[rank-axes.length];
```
Comment (PeterC-DLS) [inline org.eclipse.january/src/org/eclipse/january/dataset/LazyMaths.java]: I think SliceNDIterator.getUsedSlice().getShape() will give you this.

**Comment by tschoonj** [comment]
Done

**Comment by tschoonj** [comment]
None of the `getXXXXSlice().getShape()` calls result in the same array as `newShape`...

**Comment by PeterC-DLS** [comment]
Maybe one of `getXXXSlice().getSourceShape()` works...

**Comment by tschoonj** [comment]
Done

**Comment by tschoonj** [comment]
Thanks! Any chance this will make it into the upcoming DAWN release?

**Comment by PeterC-DLS** [comment]
No chance, I think. The plan is to release for October so next Dawn aka 2.7.

**Pull request merged**